field/field_GF2_128.cpp

```diff
@@ -235,6 +235,49 @@ bool GF2_128::operator!=(const GF2_128 &other) const {
 }
 
 GF2_128 GF2_128::inverse() const {
+  size_t u[11] = {1, 2, 3, 6, 12, 24, 48, 51, 63, 126, 127};
+  __m128i b[11];
+  size_t u_len = 11;
+
+  b[0] = *(this->as_const_m128i());
+
+  for (size_t i = 1; i < u_len; ++i) {
+    size_t p = u[i - 1];
+    size_t q = u[i] - p;
+
+    size_t p_index;
+    for (size_t m = 0; m < i; ++m) {
+      if (u[m] == p) {
+        p_index = m;
+        break;
+      }
+    }
+
+    size_t q_index;
+    for (size_t m = 0; m < i; ++m) {
+      if (u[m] == q) {
+        q_index = m;
+        break;
+      }
+    }
+
+    __m128i b_p = b[p_index];
+    __m128i b_q = b[q_index];
+
+    for (size_t m = q; m; --m) {
+      gf128sqr(&b_p, &b_p);
+    }
+
+    gf128mul(&b[i], &b_p, &b_q);
+  }
```

Comment:
In general, we can probably do away with outer loop and just fully unroll it, but it is probably fine to just let the compiler do it, we can test this later.